setup.py

```diff
@@ -364,19 +426,30 @@ def build_extension(self, ext: _BaseExtension) -> None:
         src_file: Path = ext.src_path(self)
         dst_file: Path = ext.dst_path(self)
 
-        # Ensure that the destination directory exists.
-        self.mkpath(os.fspath(dst_file.parent))
-
         # Copy the file.
-        self.copy_file(os.fspath(src_file), os.fspath(dst_file))
+        src_list = src_file.parent.rglob(src_file.name)
+        for src in src_list:
+            if ext.is_dst_dir:
+                # Destination is a prefix directory. Copy the file to the
+                # destination directory.
+                dst = dst_file / src
+            else:
+                # Destination is a file. Copy the file to the destination file.
+                dst = dst_file
+
+            # Ensure that the destination directory exists.
+            if not dst.parent.exists():
```

Comment:
Is the "not exists" part important? We didn't need it before. I think `self.mkpath` might do the same check internally